source/extensions/transport_sockets/tap/tap.h

```diff
@@ -1,22 +1,62 @@
 #pragma once
 
-#include <fstream>
-
 #include "envoy/config/transport_socket/tap/v2alpha/tap.pb.h"
-#include "envoy/data/tap/v2alpha/transport.pb.h"
+#include "envoy/data/tap/v2alpha/wrapper.pb.h"
 #include "envoy/event/timer.h"
 #include "envoy/network/transport_socket.h"
 
+#include "extensions/common/tap/extension_config_base.h"
+#include "extensions/common/tap/tap_config_base.h"
+#include "extensions/transport_sockets/tap/tap_config.h"
+
 namespace Envoy {
 namespace Extensions {
 namespace TransportSockets {
 namespace Tap {
 
+class SocketTapConfigImpl;
```

Comment:
have those two class (PerSocketTapperImpl and this) in separate file? `tap_config_impl.{h,cc}`?